### PR TITLE
Update Python version to 3.10.10

### PR DESCRIPTION
**Pull request by macmackiewicz**
Additional changes:
* fix deprecation warning by replacing pkg_resources with importlib,
* remove unused imports,
* fix import sorting

### CODE REVIEW

File: pterradactyl/facter/woops/manifest_path.py [hunk 1]
```diff
@@ -4,11 +4,15 @@
...
 # Size of source mod 2**32: 1517 bytes
-import os, re
+import os
+import re
 from argparse import Action
-from .base import BaseFacter
+
 from pterradactyl.config import Config
 
+from .base import BaseFacter
```
Comment (abdulhuq811) [inline pterradactyl/facter/woops/manifest_path.py:13]: Use absolute imports instead of relatives

File: pterradactyl/facter/regex.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
 import re
+
 import phiera
 from jinja2 import Environment
-from .base import BaseFacter
+
 from pterradactyl.config import Config
 
+from .base import BaseFacter
```
Comment (abdulhuq811) [inline pterradactyl/facter/regex.py:8]: Use absolute imports

File: pterradactyl/facter/jinja.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
+from functools import reduce
+
 from jinja2 import Environment
+
 from .base import BaseFacter
```
Comment (abdulhuq811) [inline pterradactyl/facter/jinja.py:5]: Use absolute imports

File: pterradactyl/facter/environment.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import os
+
 from .base import BaseFacter
```
Comment (abdulhuq811) [inline pterradactyl/facter/environment.py:3]: Use absolute imports

File: pterradactyl/facter/arguments.py [hunk 5]
```diff
@@ -1,11 +1,10 @@
-import os
-import re
 from argparse import Action
 
-from .base import BaseFacter
-from pterradactyl.config import Config
 from pterradactyl.util import as_list, lookup
 
+from .base import BaseFacter
```
Comment (abdulhuq811) [inline pterradactyl/facter/arguments.py:5]: Use absolute imports

File: pterradactyl/commands/lookup.py [hunk 6]
```diff
@@ -1,15 +1,11 @@
...
+import yaml
 from yaml.representer import SafeRepresenter
 
 from pterradactyl.config import Config
-from pterradactyl.facter import Facter
-from pterradactyl.validator import Validator
-from pterradactyl.terraform.terraform import Terraform
-from pterradactyl.terraform.config import TerraformConfig
 
 from .base import AbstractBaseCommand
```
Comment (abdulhuq811) [inline pterradactyl/commands/lookup.py:10]: Use absolute imports

File: pterradactyl/commands/dump.py [hunk 7]
```diff
@@ -1,9 +1,7 @@
-import os
 import yaml
 
 from pterradactyl.config import Config
 from pterradactyl.facter import Facter
-from pterradactyl.terraform.terraform import Terraform
 from pterradactyl.terraform.config import TerraformConfig
 
 from .base import AbstractBaseCommand
```
Comment (abdulhuq811) [inline pterradactyl/commands/dump.py:7]: Use absolute imports

**Review by abdulhuq811 — COMMENTED** [review]
Need to update CHANGELOG.md as well

File: pterradactyl/facter/woops/manifest_path.py (hunk 1 above)
Comment (macmackiewicz) [reply pterradactyl/facter/woops/manifest_path.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/facter/regex.py (hunk 2 above)
Comment (macmackiewicz) [reply pterradactyl/facter/regex.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/facter/environment.py (hunk 4 above)
Comment (macmackiewicz) [reply pterradactyl/facter/environment.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/facter/jinja.py (hunk 3 above)
Comment (macmackiewicz) [reply pterradactyl/facter/jinja.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/facter/arguments.py (hunk 5 above)
Comment (macmackiewicz) [reply pterradactyl/facter/arguments.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/commands/lookup.py (hunk 6 above)
Comment (macmackiewicz) [reply pterradactyl/commands/lookup.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

File: pterradactyl/commands/dump.py (hunk 7 above)
Comment (macmackiewicz) [reply pterradactyl/commands/dump.py]: done in ccd84ae80ebddd8673695dff13c7489a8f4394ff

**Review by abdulhuq811 — APPROVED** [review]
Summary: Upgrade to python3.10

**Pull request merged**